compiler/src/dotty/tools/dotc/transform/CompleteJavaEnums.scala

```diff
@@ -114,37 +113,42 @@ class CompleteJavaEnums extends MiniPhase with InfoTransformer { thisPhase =>
 
   private def isJavaEnumValueImpl(cls: Symbol)(using Context): Boolean =
     cls.isAnonymousClass
-    && ((cls.owner.name eq nme.DOLLAR_NEW) || cls.owner.isAllOf(EnumCase))
-    && {
-      val enumCls = cls.owner.owner.linkedClass
-      enumCls.derivesFromJavaEnum && enumCls.is(Enum)
-    }
+    && (((cls.owner.name eq nme.DOLLAR_NEW) && cls.owner.isAllOf(Private|Synthetic)) || cls.owner.isAllOf(EnumCase))
+    && cls.owner.owner.linkedClass.derivesFromJavaEnum
+
+  @threadUnsafe
+  private lazy val enumCaseOrdinals: mutable.Map[Symbol, Int] = mutable.AnyRefMap.empty
```

Comment:
I think `newMutableSymbolMap` can be used here instead.

Reply:
thanks :)